views/js/runner/provider/qti.js

```diff
@@ -226,7 +233,9 @@ define([
              * @param {Promise} [loadPromise] - wait this Promise to resolve before loading the item.
              */
             function computeNext(action, params, loadPromise){
-
+                var dialog = false;
```

Comment:
All that concerns this functionality can be moved to a new component. Also all constants should be configurable and came from the backend.
I mean these:
```
var echoDelayUpdate = 15;
var echoPauseLimit = 120;
var echoExceptionName = 'CatEngine';
```